example/spec/loadTest/ingestLoad_queueGranPassThruSF.js

```diff
@@ -0,0 +1,253 @@
+  }
+};
+
+describe('The Granule Ingest Load Test ', () => {
+  beforeAll(async () => {
+    if (!Number.isFinite(Number(granuleCountPerWorkflow)) || granuleCountPerWorkflow > 450) {
+      beforeAllFailed = `===== beforeAll() - Invalid input for number of granules per Workflow detected - ${granuleCountPerWorkflow} =====`;
+      throw new Error(beforeAllFailed);
+    } else {
+      try {
+        config = await loadConfig();
+
+        process.env.GranulesTable = `${config.stackName}-GranulesTable`;
+        process.env.ExecutionsTable = `${config.stackName}-ExecutionsTable`;
+        process.env.CollectionsTable = `${config.stackName}-CollectionsTable`;
+
+        await pMap(
+          new Array(totalWorkflowCount).fill().map((_, index) => index + 1),
+          async (index) => {
+            await batchGranulesProcessing(index);
+          },
+          { concurrency: 8 }
+        );
+      } catch (error) {
+        beforeAllFailed = `===== beforeAll() failed =====\n ${error}`;
+        throw new Error(beforeAllFailed);
+      }
+    }
+  });
+
+  afterAll(async () => {
+    // clean up stack state added by test
+    await Promise.all(
+      totalInputPayloads.map(async (inPayload) =>
+        await cleanupLoadTestGranules(config.stackName, inPayload.granules))
+    );
+
+    console.log(blueConsoleLog(), '\n===== Delete lambdaOutputPayload ExecutionArns =====');
+    queueGranulesChildExecutionArns.forEach((lambdaOutputPayload) => {
+      lambdaOutputPayload.running.forEach((childExecutionArn) => {
+        deleteExecution({
+          prefix: config.stackName,
+          executionArn: childExecutionArn,
+        });
+        // console.log(blueConsoleLog(), `${JSON.stringify(childExecutionArn)}`);
+      });
+    });
+
+    if (queueGranulesExecutionArns !== undefined) {
+      console.log(blueConsoleLog(), '\n===== Delete queueGranules ExecutionArns =====');
+      queueGranulesExecutionArns.forEach((qGranulesExecutionArn) => {
+        deleteExecution({
+          prefix: config.stackName,
+          executionArn: qGranulesExecutionArn,
+        });
+        console.log(blueConsoleLog(), `${JSON.stringify(qGranulesExecutionArn)}`);
+      });
+    }
+
+    await Promise.all(
+      testSuffixes.map(async (testSuffix) => {
+        await cleanupCollections(
+          config.stackName,
+          config.bucket,
+          collectionsDir,
+          testSuffix
+        );
+        cleanupProviders(
+          config.stackName,
+          config.bucket,
+          providersDir,
+          testSuffix
+        );
+      })
+    );
+  });
+
+  it('writes to database the expected number of granules with status completed', () => {
+    const expectedGranuleCount = granuleCountPerWorkflow * totalWorkflowCount;
+    expect(expectedGranuleCount * granuleCountThreshold < totalGranulesCompleted.length && totalGranulesCompleted.length <= expectedGranuleCount).toBeTruthy();
```

Comment:
Can you explain what this expectation is asserting? What's the reasoning behind `granuleCountThreshold`? Why is OK for `totalGranulesCompleted.length <= expectedGranuleCount`? Shouldn't we have a problem if `totalGranulesCompleted` is less than `expectedGranuleCount`?

Reply:
This means to “allow” 5% room for ingestion failure/timeout since we are working with a large volume of granules, taking into account the AWS resource availability at runtime. So long as the system load ingest capability can be validated, the 95% round about number should suffice. This idea was actually "derived" from an existing load test previously written by the one and only Jonathan.